### PR TITLE
Fix unpublished mods being displayed on frontend

### LINKED ISSUE #189: The unpublished mods are visible to the public
Steps to reproduce:
* Create an unpublished mod.
* Go to the author's page.

Observed result:
* The unpublished mod is visible in the "Mods by \<username\>" section.
* When clicked on the mod, the result is "Unauthorized", which is the right behavior.

Expected result:
* The unpublished mod must only be visible to the owner.

### LINKED ISSUE #228: Internal Server Error when accessing user profiles
**Description (What went wrong?):**
Some user profiles return Internal Server Error, with no custom style. Just a plain HTML header saying "Internal Server Error".

**Reproduction Steps (What did you do?):**,
I tried to create a mod for No Man's Sky, to test an old bug that unpublished mods are visible. After creating the mod, I couldn't view my profile and I couldn't publish the mod either.

Profile link: https://spacedock.info/profile/Xinayder

**Expected Behavior (What do you think should have happened instead?):**
Profile page should work.

**Environment (OS/Browser/Plugins/etc):**
- Firefox v68.0.1 (64-bit)
- Arch Linux, kernel 5.2.6-arch1-1-fsync
- uBlock Origin, HTTPS Everywhere, Privacy Badger (none interfere with SpaceDock)

**Extra Information (Screenshots/Error Messages/Javascript Console Output):**
Accessing the same page with a private tab has the same error, but with SpaceDock's styling.

### PR DESCRIPTION
**Pull request by RockyTV**
Closes #189

### CODE REVIEW

**Comment by RockyTV** [comment]
do'nt merge yet

**Comment by DasSkelett** [comment]
But actually, this hides them also for the mod author himself, even if he's logged in. Is there a way to show them in this case?

**Comment by allista** [comment]
This change can't close the  #189, as it's purely API bug, while this fixes templates.

**Comment by RockyTV** [comment]
Apparently it's already handled by the API, so I'm closing this.

https://github.com/KSP-SpaceDock/SpaceDock/blob/alpha/KerbalStuff/blueprints/api.py#L351

**Comment by RockyTV** [comment]
After reviewing it again, I figured out what's the problem. The current layout was accessing all the mods the user created, but the backend exposes a list of created mods that filters unpublished mods and sorts the rest. Same goes for followed mods.

File: templates/view_profile.html [hunk 1]
```diff
@@ -156,29 +156,29 @@ <h1>Mod Packs by {{ profile.username }}</h1>
 </div>
 {% endif %}
 
-{% if len(profile.mods) != 0 %}
+{% if len(mods_created) != 0 %}
```
Comment (HebaruSan) [inline templates/view_profile.html]: `mods_created` is the same as `profile.mods`, but sorted. How would this change filter out unpublished mods?

https://github.com/KSP-SpaceDock/SpaceDock/blob/c6ed96ddf04302cae66153cf8b66e16624db6eb9/KerbalStuff/blueprints/profile.py#L20

**Pull request merged**